include/xtensor/xbuilder.hpp

```diff
@@ -738,16 +738,11 @@ namespace xt
 
         std::size_t dim_1 = shape[axis_1];
         std::size_t dim_2 = shape[axis_2];
-
         std::size_t n_dim = arr.dimension();
 
         offset >= 0 ? dim_2 -= offset : dim_1 += offset;
 
         auto diag_size = dim_2 < dim_1 ? dim_2 : dim_1;
-        if (diag_size < 0)
-        {
```

Comment:
We should probably still use an unsigned integer and use std max or std min in the following lines.